test/javascripts/lib/utilities-test.js.es6

```diff
@@ -27,10 +27,13 @@ test("urlValid", function() {
   ok(urlValid('http://meta.discourse.org/'), "allows http url");
   ok(urlValid('http://meta.discourse.org'), "allows url without trailing slash");
   ok(urlValid('https://meta.discourse.org/'), "allows https url");
+  ok(urlValid('http://142.42.1.1:8080/'), "allows ip with port");
+  ok(urlValid("http://-.~_!$&'()*+,;=:%40:80%2f::::::@example.com"));
```

Comment:
Yes, it is, your browser should redirect you to `example.com`. Those characters are just user information. I think I can just change them to a dashed domain name. Would be easier to understand